src/app/api/routes/accesses.py

```diff
@@ -2,19 +2,28 @@
 from fastapi import APIRouter, Path, HTTPException
 from app.api import crud, security
 from app.db import accesses
-from app.api.schemas import AccessBase, AccessRead, AccessAuth, AccessCreation, Cred, CredHash
+from app.api.schemas import AccessBase, AccessRead, AccessAuth, AccessCreation, Cred, CredHash, Login
 
 
 router = APIRouter()
 
 
-async def post_access(login: str, password: str, scopes: str) -> AccessRead:
+async def update_access_login(login: str, access_id: int):
+    """ Assume access_id exists and login does not exist elsewhere"""
+    return await crud.update_entry(accesses, Login(login=login), access_id)
+
+
+async def check_for_access_login_existence(login: str):
     # Check that the login does not already exist
     if await crud.fetch_one(accesses, {'login': login}) is not None:
         raise HTTPException(
             status_code=400,
             detail=f"An entry with login='{login}' already exists.",
         )
```

Comment:
I would suggest to return the fetched entry, you will be able to use that in devices & users

Reply:
Well we hope the fetched entry is None; otherwise an exception is raised.

Reply:
my bad, late review :sweat_smile: